app/src/main/java/io/homeassistant/companion/android/onboarding/integration/MobileAppIntegrationFragment.kt

```diff
@@ -171,32 +170,39 @@ class MobileAppIntegrationFragment : Fragment(), MobileAppIntegrationView {
         dialog?.dismiss()
 
         if (permissions.contains(Manifest.permission.ACCESS_FINE_LOCATION) &&
-            Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
+            Build.VERSION.SDK_INT >= Build.VERSION_CODES.R
+        ) {
             requestPermissions(arrayOf(Manifest.permission.ACCESS_BACKGROUND_LOCATION), LOCATION_REQUEST_CODE)
         }
 
         if (requestCode == LOCATION_REQUEST_CODE) {
             val hasPermission = grantResults.all { it == PackageManager.PERMISSION_GRANTED }
-            locationTracking.isChecked = hasPermission
-            setLocationTracking(hasPermission)
+            setLocationTracking(locationTracking, hasPermission)
             requestBackgroundAccess()
         }
     }
 
-    private fun setLocationTracking(enabled: Boolean) {
+    private fun setLocationTracking(locationTrackingSwitch: SwitchMaterial, enabled: Boolean) {
+        locationTrackingSwitch.isChecked = enabled
+
         val sensorDao = AppDatabase.getInstance(requireContext()).sensorDao()
         arrayOf(
             LocationSensorManager.backgroundLocation,
             LocationSensorManager.zoneLocation,
             LocationSensorManager.singleAccurateLocation
         ).forEach { basicSensor ->
             var sensorEntity = sensorDao.get(basicSensor.id)
+            var sensorEnabled = enabled
+            if (basicSensor.id == LocationSensorManager.singleAccurateLocation.id && !DisabledLocationHandler.hasGPS(requireContext())) {
+                sensorEnabled = false
+            }
```

Comment:
Do we need this here if we aren't even showing the sensor when they don't have GPS?

Reply:
Thanks for the review 👍  No, not really. I just wanted to keep the database consistent with the display. I will remove it.